src/System.Linq.Expressions/tests/Unary/UnaryConvertTests.cs

```diff
@@ -0,0 +1,391 @@
+            {
+                VerifyUnaryConvert(e);
+            }
+        }
+
+        [Fact]
+        public static void ConvertUnboxingTest()
+        {
+            foreach (var e in ConvertUnboxing())
+            {
+                VerifyUnaryConvert(e);
+            }
+        }
+
+        [Fact]
+        public static void ConvertUnboxingInvalidCastTest()
+        {
+            foreach (var e in ConvertUnboxingInvalidCast())
+            {
+                VerifyUnaryConvertThrows<InvalidCastException>(e);
+            }
+        }
+
+        [Fact] // [Issue(4019, "https://github.com/dotnet/corefx/issues/4019")]
+        public static void CheckUnaryConvertBooleanToNumericTest()
+        {
+            foreach (var kv in ConvertBooleanToNumeric())
+            {
+                VerifyUnaryConvert(kv.Key, kv.Value);
+            }
+        }
+
+        [Fact] // [Issue(4019, "https://github.com/dotnet/corefx/issues/4019")]
+        public static void ConvertNullToNonNullableValueTest()
+        {
+            foreach (var e in ConvertNullToNonNullableValue())
+            {
+                VerifyUnaryConvertThrows<NullReferenceException>(e);
+            }
+        }
+
+        [Fact] // [Issue(4019, "https://github.com/dotnet/corefx/issues/4019")]
+        public static void ConvertNullToNullableValueTest()
+        {
+            foreach (var e in ConvertNullToNullableValue())
+            {
+                VerifyUnaryConvert(e, null);
+            }
+        }
+
+        [Fact] // [Issue(4019, "https://github.com/dotnet/corefx/issues/4019")]
+        public static void ConvertUnderlyingTypeToEnumTypeTest()
+        {
+            var enumValue = DayOfWeek.Monday;
+            var value = (int)enumValue;
+
+            foreach (var o in new[] { Expression.Constant(value, typeof(int)), Expression.Constant(value, typeof(ValueType)), Expression.Constant(value, typeof(object)) })
+            {
+                VerifyUnaryConvert(Expression.Convert(o, typeof(DayOfWeek)), enumValue);
+            }
+        }
+
+        [Fact] // [Issue(4019, "https://github.com/dotnet/corefx/issues/4019")]
+        public static void ConvertUnderlyingTypeToNullableEnumTypeTest()
+        {
+            var enumValue = DayOfWeek.Monday;
+            var value = (int)enumValue;
+
+            var cInt = Expression.Constant(value, typeof(int));
+            VerifyUnaryConvert(Expression.Convert(cInt, typeof(DayOfWeek?)), enumValue);
+
+            var cObj = Expression.Constant(value, typeof(object));
+            VerifyUnaryConvertThrows<InvalidCastException>(Expression.Convert(cObj, typeof(DayOfWeek?)));
+        }
+
+        [Fact] // [Issue(4019, "https://github.com/dotnet/corefx/issues/4019")]
+        public static void ConvertArrayToIncompatibleTypeTest()
+        {
+            var arr = new object[] { "bar" };
+
+            foreach (var t in new[] { typeof(string[]), typeof(IEnumerable<char>[]) })
```

Comment:
Would it make sense to do this as a [Theory] with [InlineData(typeof(string[])] etc. instead of with a loop?